woocommerce/woocommerce-gutenberg-products-block
# Add context support to Atomic Blocks

#2399 is a mammoth PR so I'm breaking out some fixes and changes to review separately.

This PR makes our Atomic Blocks context ready. The context support is backwards compatible with the old way of passing a product as a prop - that takes priority if used.

Changes:

- Refactored Atomic Blocks to not break when no product is given - this is important whilst context is loaded.
- Renamed `useProductDataContextContext` to `useProductDataContext`
- Renamed `useInnerBlockConfigurationContext` to `useInnerBlockLayoutContext` to better reflect its use.
- Added context handling to each Atomic Block
- Removed default "preview" data and instead injected it via the new context!

### How to test the changes in this Pull Request:

1. Using All Products Block, check there are no regressions in the layout edit view, in particular, check the preview data is shown correctly since this is now coming from context.
2. Check frontend view of All Products for regressions. Add to cart buttons etc.

That should cover it. Nothing else is using context except the layout edit preview data, so just testing all works as before is all thats needed.
